lib/net/nvnetdrv/nvnetdrv.c

```diff
@@ -79,37 +79,34 @@ static KINTERRUPT g_interrupt;
 static HANDLE g_irqThread;
 static KEVENT g_irqEvent;
 
-// Manage RX ring
-static volatile struct descriptor_t *g_rxRing;
-static KSEMAPHORE g_rxRingFreeDescriptors;
-static HANDLE g_rxRingRequeueThread;
+// Manage RX and TX rings
+static struct descriptor_t *g_rxRing;
+static struct descriptor_t *g_txRing;

```

Comment:
Hm, I don't think we can drop the `volatile` here as the hw modifies the `flags` member. Interestingly this change doesn't lead to a bug even with LTO, probably because `nvnetdrv_handle_rx_irq` copies the address into a proper volatile pointer.

Imho the initialization of these should be changed, too, as it's currently casting to a non-volatile pointer, if you want to squeeze that into one of the commits in this PR:
```
// Setup the TX and RX ring descriptor pointers
    g_rxRing = (struct descriptor_t *)descriptors;
    g_txRing = (struct descriptor_t *)descriptors + RX_RING_SIZE;
```

Reply:
ah yep gotcha, fixed thanks